application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,220 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.BotCommandAdapter;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.MessageContextCommand;
+import org.togetherjava.tjbot.features.utils.StringDistances;
+
+import java.awt.Color;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+public final class TransferQuestionCommand extends BotCommandAdapter
+        implements MessageContextCommand {
+    private static final Logger logger = LoggerFactory.getLogger(TransferQuestionCommand.class);
+    private static final String COMMAND_NAME = "transfer-question";
+    private static final String TRANSFER_QUESTION_TITLE_ID = "transferID";
+    private static final String TRANSFER_QUESTION_INPUT_ID = "transferQuestion";
+    private static final String TRANSFER_QUESTION_TAG = "tags";
+    private static final int TITLE_MAX_LENGTH = 50;
+    private static final Pattern TITLE_COMPACT_REMOVAL_PATTERN = Pattern.compile("\\W");
+    private static final int TITLE_COMPACT_LENGTH_MIN = 2;
+    private static final int TITLE_COMPACT_LENGTH_MAX = 30;
+    private final Predicate<String> isHelpForumName;
+    private final List<String> defaultTags;
+    private static final Color EMBED_COLOR = new Color(50, 164, 168);
+
+
+    /**
+     * Creates a new instance.
+     */
+    public TransferQuestionCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+        isHelpForumName =
+                Pattern.compile(config.getHelpSystem().getHelpForumPattern()).asMatchPredicate();
+
+        defaultTags = config.getHelpSystem().getCategories();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        String originalMessage = event.getTarget().getContentRaw();
+        String originalMessageId = event.getTarget().getId();
+        String originalChannelId = event.getChannel().getId();
+        String authorId = event.getTarget().getAuthor().getId();
+
+        TextInput transferQuestionTitle =
+                TextInput.create(TRANSFER_QUESTION_TITLE_ID, "Title", TextInputStyle.SHORT)
+                    .setMaxLength(70)
+                    .setMinLength(8)
+                    .setValue(createTitle(originalMessage))
+                    .build();
+
+        TextInput transferQuestionInput = TextInput
+            .create(TRANSFER_QUESTION_INPUT_ID, "Transfer question menu", TextInputStyle.PARAGRAPH)
+            .setValue(originalMessage)
+            .setRequiredRange(3, 2000)
+            .build();
+
+        TextInput transferQuestionTag = TextInput
+            .create(TRANSFER_QUESTION_TAG, "Transfer question tags", TextInputStyle.SHORT)
+            .setValue("Java")
```

Comment:
i still see java here

Reply:
Because it's supposed to be there, the idea is Java is the most popular tag so 90% of the times you won't need to change the tag but if you do it can be done and will find the closest match regardless.

Reply:
the idea was to pull the first tag from config and the use ped on submit

Reply:
I don't see how it helps, it's a modal field supposed to be prefilled for convenience. It can be changed or not.

Reply:
taz is right on this. please take the first entry from the config instead.

u have to think about other servers as well. for example, TJ-Bot is used on my kotlin server as well. and the top tag there is `Kotlin`.